types/versions/compare.go

```diff
@@ -0,0 +1,62 @@
+package versions
+
+import (
+	"strconv"
+	"strings"
+)
+
+// compare compares two version strings
+// returns -1 if v1 < v2, 1 if v1 > v2, 0 otherwise.
+func compare(v1, v2 string) int {
```

Comment:
this is like the same as https://github.com/docker/docker/blob/master/pkg/version/version.go#L11